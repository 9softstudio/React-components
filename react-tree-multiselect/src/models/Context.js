import React from 'react';

export default React.createContext({
    rootNode: {},
    nodes: [],
    onToggle: () => { },
    onCheck: () => { }
});