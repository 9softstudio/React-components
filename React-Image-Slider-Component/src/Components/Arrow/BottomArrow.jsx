import React, { Component } from 'react';
import Context from '../../slider-context';

export default class BottomArrow extends Component {
    render() {
        return (
            <Context.Consumer>{
                ({ onBottomArrowClick }) =>
                    <img src="img/slider-bottom-arrow.png" className="slider-arrow-position bottom-arrow" onClick={onBottomArrowClick} />
            }</Context.Consumer>);
    }
};