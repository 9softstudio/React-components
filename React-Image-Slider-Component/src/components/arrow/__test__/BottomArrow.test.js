import React from 'react';
import BottomArrow from '../BottomArrow';
import Context from '../../../slider-context';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const component = renderer
        .create(
            <Context.Provider value={({ onBottomArrowClick: jest.fn() })}>
                <BottomArrow />
            </Context.Provider>
        );
    expect(component.toJSON()).toMatchSnapshot();
});

test('on click: onBottomArrowClick()', () => {
    const onBottomArrowClick = jest.fn();
    const component = renderer
        .create(
            <Context.Provider value={({ onBottomArrowClick: onBottomArrowClick })}>
                <BottomArrow />
            </Context.Provider>
        );
    expect(component.toJSON().props.onClick).toEqual(onBottomArrowClick);
});