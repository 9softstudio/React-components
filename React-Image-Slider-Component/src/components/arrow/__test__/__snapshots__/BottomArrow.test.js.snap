// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  className="slider-arrow-position bottom-arrow"
  onClick={[MockFunction]}
  src="img/slider-bottom-arrow.png"
/>
`;
